integration-tests/q-agentic-chat-server/src/tests/agenticChatInteg.test.ts

```diff
@@ -113,20 +125,248 @@ describe('Q Agentic Chat Server Integration Tests', async () => {
         expect(result.capabilities).to.exist
     })
 
+    beforeEach(() => {
+        tabId = crypto.randomUUID()
+        partialResultToken = crypto.randomUUID()
+    })
+
     after(async () => {
         client.exit()
     })
 
+    afterEach(function (this: Mocha.Context) {
+        if (this.currentTest?.state === 'failed') {
+            console.log('\n=== SERVER LOGS ON FAILURE ===')
+            console.log(serverLogs.join(''))
+            console.log('=== END SERVER LOGS ===\n')
+        }
+        serverLogs = []
+    })
+
     it('responds to chat prompt', async () => {
         const encryptedMessage = await encryptObjectWithKey<ChatParams>(
-            { tabId: 'tab-id', prompt: { prompt: 'Hello' } },
+            { tabId, prompt: { prompt: 'Hello' } },
             encryptionKey
         )
-        const result = await endpoint.send('aws/chat/sendChatPrompt', { message: encryptedMessage })
+        const result = await client.sendChatPrompt({ message: encryptedMessage })
         const decryptedResult = await decryptObjectWithKey<ChatResult>(result, encryptionKey)
 
         expect(decryptedResult).to.have.property('messageId')
         expect(decryptedResult).to.have.property('body')
         expect(decryptedResult.body).to.not.be.empty
     })
+
+    it('reads file contents using fsRead tool', async () => {
+        const encryptedMessage = await encryptObjectWithKey<ChatParams>(
+            {
+                tabId,
+                prompt: { prompt: 'Read the contents of the test.py file using the fsRead tool.' },
```

Comment:
does test.py exist?

Reply:
yes it exists in testFixtures folder with some dummy code